test/unit/array.spec.ts

```diff
@@ -226,3 +226,36 @@ test.each([`["foo", "bar"].length`, `["foo", "bar"][0]`, `[() => "foo", () => "b
         expect(util.transpileAndExecute(code)).toBe(expectResult);
     }
 );
+
+const genericChecks = [
+    "function generic<T extends number[]>(array: T)",
+    "function generic<T extends [...number[]]>(array: T)",
+    "function generic<T extends any>(array: T[])",
+    "type ArrayType = number[]; function generic<T extends ArrayType>(array: T)",
+];
```

Comment:
Few cases that may worth to be tested:

```suggestion
    "function generic<T extends number[]>(array: T & {})",
    "function generic<T extends number[] & {}>(array: T)",
];
```